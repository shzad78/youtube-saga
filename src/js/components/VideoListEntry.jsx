import React from 'react';
import PropTypes from 'prop-types';

const VideoListEntry = ({video, handleSelectedVideo}) => {

  return (
    <div className="videoListEntry"> 
      <div className="media">
        <div className="media-left media-middle">
          <img 
            className="media-object" 
            src={video.snippet.thumbnails.default.url} 
          />
        </div>
        <div className="media-body">
          <h4 
            className="media-heading"
            onClick={() => handleSelectedVideo(video)}
          >
            {video.snippet.title}
          </h4>
          <p className="media-description">{video.snippet.description}</p>
        </div>
      </div>
    </div>
  )
};

VideoListEntry.propTypes = {
  video: PropTypes.object.isRequired,
  handleSelectedVideo: PropTypes.func.isRequired,
}

export default VideoListEntry;

