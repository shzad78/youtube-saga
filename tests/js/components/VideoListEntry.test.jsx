import React from 'react';
import VideoListEntry from '../../../src/js/components/VideoListEntry';

//Below are related to enzyme
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });


describe('Testing VideoListEntry Component',()=>{
  const handleSelectedVideo = jest.fn();
  const video = {
    snippet: {
      title: 'some title',
      description: 'some description',
      thumbnails: {
        default: {
          url: 'someurl'
        }
      }
    }
  }

  const wrapper = shallow(
    <VideoListEntry 
      handleSelectedVideo={handleSelectedVideo} 
      video={video}
    />
  );

  it('renders the VideoListEntry Component', ()=>{  
    expect(wrapper).toMatchSnapshot();
  });
  
  it('runs handleSelectedVideo function with video arg when clicked', ()=>{
    wrapper.find('.media-heading').simulate('click')
    expect(handleSelectedVideo.mock.calls[0]).toEqual([video]);
  });

});

