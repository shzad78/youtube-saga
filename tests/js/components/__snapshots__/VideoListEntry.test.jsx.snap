// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoListEntry Component renders the VideoListEntry Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    video={
      Object {
        "snippet": Object {
          "description": "some description",
          "thumbnails": Object {
            "default": Object {
              "url": "someurl",
            },
          },
          "title": "some title",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="media"
      >
        <div
          className="media-left media-middle"
        >
          <img
            className="media-object"
            src="someurl"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="media-heading"
            onClick={[Function]}
          >
            some title
          </h4>
          <p
            className="media-description"
          >
            some description
          </p>
        </div>
      </div>,
      "className": "videoListEntry",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="media-left media-middle"
          >
            <img
              className="media-object"
              src="someurl"
            />
          </div>,
          <div
            className="media-body"
          >
            <h4
              className="media-heading"
              onClick={[Function]}
            >
              some title
            </h4>
            <p
              className="media-description"
            >
              some description
            </p>
          </div>,
        ],
        "className": "media",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="media-object"
              src="someurl"
            />,
            "className": "media-left media-middle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "media-object",
              "src": "someurl",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="media-heading"
                onClick={[Function]}
              >
                some title
              </h4>,
              <p
                className="media-description"
              >
                some description
              </p>,
            ],
            "className": "media-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "some title",
                "className": "media-heading",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "some title",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "some description",
                "className": "media-description",
              },
              "ref": null,
              "rendered": "some description",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="media"
        >
          <div
            className="media-left media-middle"
          >
            <img
              className="media-object"
              src="someurl"
            />
          </div>
          <div
            className="media-body"
          >
            <h4
              className="media-heading"
              onClick={[Function]}
            >
              some title
            </h4>
            <p
              className="media-description"
            >
              some description
            </p>
          </div>
        </div>,
        "className": "videoListEntry",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="media-left media-middle"
            >
              <img
                className="media-object"
                src="someurl"
              />
            </div>,
            <div
              className="media-body"
            >
              <h4
                className="media-heading"
                onClick={[Function]}
              >
                some title
              </h4>
              <p
                className="media-description"
              >
                some description
              </p>
            </div>,
          ],
          "className": "media",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="media-object"
                src="someurl"
              />,
              "className": "media-left media-middle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "media-object",
                "src": "someurl",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  className="media-heading"
                  onClick={[Function]}
                >
                  some title
                </h4>,
                <p
                  className="media-description"
                >
                  some description
                </p>,
              ],
              "className": "media-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "some title",
                  "className": "media-heading",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "some title",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "some description",
                  "className": "media-description",
                },
                "ref": null,
                "rendered": "some description",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
